src/data/finsupp/basic.lean

```diff
@@ -699,6 +699,13 @@ lemma on_finset_prod {s : finset α} {f : α → M} {g : α → M → N}
   (on_finset s f hf).prod g = ∏ a in s, g a (f a) :=
 finset.prod_subset support_on_finset_subset $ by simp [*] { contextual := tt }
 
+@[to_additive] lemma prod_eq_prod_fintype {N β : Type*} [comm_monoid N] [fintype α]
+  [has_zero β] (g : α → β → N) (h : ∀ a, g a 0 = 1) (f : α →₀ β) :
+  f.prod g = ∏ a:α, g a (f a) :=
```

Comment:
I think you can eliminate the type here if you want:
```suggestion
  f.prod g = ∏ a, g a (f a) :=
```

Reply:
It would work, but I find it a bit clearer this way, if it is possible.

Reply:
In that case let's use the normal spacing convention:
```suggestion
  f.prod g = ∏ a : α, g a (f a) :=
```